test/rpc_test.py

```diff
@@ -1098,12 +1098,20 @@ def _test_rref_leak(self, ignore_leak=False):
             import torch.distributed.rpc.api as api
```

Comment:
Will the following be more laconic?

```python
if ignore_leak:
  import torch.distributed.rpc.api as api
  api._ignore_rref_leak = True
  rpc.shutdown(graceful=False)
else:
  assert_context = self.assertRaisesRegex(RuntimeError, "Leaking RRef")
  assert_context.__enter__()
  rpc.shutdown(graceful=False)
  if not assert_context.__exit__(*sys.exc_info()):
    raise
```

Reply:
This way, we can use the with statement syntax. Other comments are not relevant if I use with statement.